File: archive/2020f/solutions/ClimbingTrees.py
from sys import stdin, stdout
print, input = stdout.write, stdin.readline

inp = input().split()
n, k = int(inp[0]), int(inp[1])

a = [int(x) for x in input().split()]
a.sort()

bef = -k*2
best = 0
curr = 0
for x in a:
    if x <= bef + k:
        curr += x
    else:
        curr = x
    best = max(best, curr)
    bef = x

print(str(best))

File: archive/2020f/solutions/ClimbingTrees.java
import java.util.*;
import java.io.*;

// By Ayush Varshney
public class ClimbingTrees {

	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt(), k = sc.nextInt();
		Long[] in = new Long[n];
		for (int i = 0; i < n; i++) {
			in[i] = sc.nextLong();
		}
		Arrays.sort(in);
		long curr = in[0], best = 0;
		for (int i = 0; i < n - 1; i++) {
			if (in[i + 1] - in[i] > k) {
				best = Math.max(curr, best);
				curr = in[i + 1];
			} else {
				curr += in[i + 1];
			}
		}
		best = Math.max(curr, best);
		System.out.println(best);
		sc.close();
	}

}